tools/gulp/packaging/build-functions.ts

```diff
@@ -0,0 +1,86 @@
+import {join} from 'path';
+import {createRollupBundle} from './rollup-helper';
+import {inlinePackageMetadataFiles} from './inline-resources';
+import {transpileFile} from './ts-compiler';
+import {ScriptTarget, ModuleKind} from 'typescript';
+import {DIST_BUNDLES, DIST_ROOT, SOURCE_ROOT, PROJECT_ROOT} from '../constants';
+import {copyFiles} from '../util/copy-files';
+import {
+  addPureAnnotationsToFile, createMetadataFile, createTypingFile, remapSourcemap, uglifyFile,
+  updatePackageVersion
+} from './build-utils';
+
+// There are no type definitions available for these imports.
+const uglify = require('uglify-js');
+const sorcery = require('sorcery');
+
+/**
+ * Copies different output files into a folder structure that follows the `angular/angular`
+ * release folder structure. The output will also contain a README and the according package.json
+ * file. Additionally the package will be Closure Compiler and AOT compatible.
+ */
+export function composeRelease(packageName: string) {
+  // To avoid refactoring of the project the package material will map to the source path `lib/`.
+  const sourcePath = join(SOURCE_ROOT, packageName === 'material' ? 'lib' : packageName);
+  const packagePath = join(DIST_ROOT, 'packages', packageName);
+  const releasePath = join(DIST_ROOT, 'releases', packageName);
+
+  inlinePackageMetadataFiles(packagePath);
+
+  copyFiles(packagePath, '**/*.+(d.ts|metadata.json)', join(releasePath, 'typings'));
+  copyFiles(DIST_BUNDLES, `${packageName}.umd?(.min).js?(.map)`, join(releasePath, 'bundles'));
+  copyFiles(DIST_BUNDLES, `${packageName}?(.es5).js?(.map)`, join(releasePath, '@angular'));
+  copyFiles(PROJECT_ROOT, 'LICENSE', releasePath);
+  copyFiles(SOURCE_ROOT, 'README.md', releasePath);
+  copyFiles(sourcePath, 'package.json', releasePath);
+
+  updatePackageVersion(releasePath);
+  createTypingFile(releasePath, packageName);
+  createMetadataFile(releasePath, packageName);
+  addPureAnnotationsToFile(join(releasePath, '@angular', `${packageName}.es5.js`));
+}
+
+/** Builds the bundles for the specified package. */
+export async function buildPackageBundles(entryFile: string, packageName: string) {
```

Comment:
I like that idea. Do you have any good ideas for the file names?

- `build-release.ts`
- `build-normal.ts`

Something like that?